cmd/report.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"runtime"
+	"strings"
+
+	"go.k6.io/k6/execution"
+	"go.k6.io/k6/lib/consts"
+)
+
+type report struct {
+	Version    string         `json:"k6_version"`
+	Executors  map[string]int `json:"executors"`
+	VUsMax     int64          `json:"vus_max"`
+	Iterations uint64         `json:"iterations"`
+	Duration   string         `json:"duration"`
+	GoOS       string         `json:"goos"`
+	GoArch     string         `json:"goarch"`
+	Modules    []string       `json:"modules"`
+	Outputs    []string       `json:"outputs"`
+}
+
+func createReport(execScheduler *execution.Scheduler, importedModules []string, outputs []string) report {
+	executors := make(map[string]int)
+	for _, ec := range execScheduler.GetExecutorConfigs() {
+		executors[ec.GetType()]++
+	}
+
+	// collect the report only with k6 public modules
+	publicModules := make([]string, 0, len(importedModules))
+	for _, module := range importedModules {
+		// Exclude JS modules extensions to prevent to leak
+		// any user's custom extensions
+		if strings.HasPrefix(module, "k6/x") {
+			continue
+		}
+		// Exclude any import not starting with the k6 prefix
+		// that identifies a k6 built-in stable or experimental module.
+		// For example, it doesn't include any modules imported from the file system.
+		if !strings.HasPrefix(module, "k6") {
+			continue
+		}
+		publicModules = append(publicModules, module)
+	}
+
+	builtinOutputs := map[string]bool{
+		"cloud": true, "csv": true, "experimental-prometheus-rw": true,
```

Comment:
Nit: These output keys (cloud, csv, etc.) come from the map https://github.com/grafana/k6/blob/master/cmd/outputs.go#L26-L47. It's better to extract this as the variable/method and use it here. That way, we protect the future maintainers/contributors from forgetting to update this place. 

Some alternatives (e.g., using `const`) could also be considered, but the main idea is to couple the places through the code, not the string values.

Reply:
Added a new proposal for addressing this in the last commits